drivers/flash/flash_stm32_qspi.c

```diff
@@ -956,9 +970,10 @@ static int spi_nor_process_bfp(const struct device *dev,
 			 * portion of flash description register 16 indicates
 			 * if it is enough to use 0xB7 instruction without
 			 * write enable to switch to 4 bytes addressing mode.
+			 * If bit 1 is set, a write enable is needed.
 			 */
-			if (dw16.enter_4ba & 0x1) {
-				rc = qspi_program_addr_4b(dev);
+			if (dw16.enter_4ba & 0x3) {
```

Comment:
This is slightly different from commit description.
With this change 4 byte address node is enabled only if bit 1 and 0 are set. From the description provided in #57498, it's not clear to me if this is what is requested by jedec spec.

Reply:
No, the "if" expression is true, if at least bit 0 or bit 1 is 1, it is not necessary that both bits are set. This is the JEDEC description:
![Screenshot_20230503_172829](https://user-images.githubusercontent.com/55055211/235964585-10db17a2-7aeb-44b2-94c0-1e9a4b87b8c8.png)
So the code is right. The specification is a bit confusing, looks like it is all exclusive, because of the "x" don't care other bits, but obviously the 0x36 I read is not all exclusive or. But makes sense I guess, because a general 4 byte address mode doesn't preclude a special instruction (bit 5).